bitmovin-analytics-datasource/src/components/FilterRow.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useState } from 'react';
+import { difference, isEmpty } from 'lodash';
+import { SelectableValue } from '@grafana/data';
+import { Box, HorizontalGroup, IconButton, InlineLabel, VerticalGroup } from '@grafana/ui';
+
+import { QueryFilter, QueryFilterOperator, QueryFilterValue } from '../types/queryFilter';
+import { QueryAdAttribute, SELECTABLE_QUERY_AD_ATTRIBUTES } from '../types/queryAdAttributes';
+import { QueryAttribute, SELECTABLE_QUERY_ATTRIBUTES } from '../types/queryAttributes';
+import { FilterInput } from './FilterInput';
+import { convertFilterValueToProperType } from '../utils/filterUtils';
+
+type Props = {
+  readonly isAdAnalytics: boolean;
+  readonly onChange: (newFilters: QueryFilter[]) => void;
+};
+
+type Filter = {
+  selectedAttribute: SelectableValue<QueryAdAttribute | QueryAttribute>;
+  selectedOperator: SelectableValue<QueryFilterOperator>;
+  rawFilterValue: string;
+  convertedFilterValue: QueryFilterValue;
+  parsingValueError: string;
+};
+
+export function FilterRow(props: Props) {
+  const [filters, setFilters] = useState<Filter[]>([]);
+
+  const mapFilterAttributesToSelectableValue = (): Array<SelectableValue<QueryAttribute | QueryAdAttribute>> => {
```

Comment:
extract out as utility
```suggestion
  const mapFilterAttributesToSelectableValue = (filters: Filter[]): Array<SelectableValue<QueryAttribute | QueryAdAttribute>> => {
```